File: @LV1/1.7/maptiles2/maptiles2.py
s = [int(x) for x in input()]

l = len(s)-1

ans_x = 0
ans_y = 0
for i, c in enumerate(s):
    if c == 1 or c == 3:
        ans_x |= (1 << (l-i))
    if c == 2 or c == 3:
        ans_y |= (1 << (l-i))

print(len(s), ans_x, ans_y)

File: LV1/1.7/maptiles2/maptiles2.rb
s = gets.chomp.split('').map(&:to_i)

l = s.size - 1

ans_x = 0
ans_y = 0

s.each_with_index do |c, i|
  ans_x |= (1 << (l - i)) if [1, 3].include?(c)
  ans_y |= (1 << (l - i)) if [2, 3].include?(c)
end

puts "#{s.size} #{ans_x} #{ans_y}"
